### PR TITLE
PWA-1689: Filtering Product by the attribute value

### PR DESCRIPTION
**Pull request by michaelyu0123**
- Updated styling for filters

## Description

In scope: Text, Content, Boolean, Multi-select and Date

As a Merchant I want to show product attributes with values on PWA Storefront using native capabilities so that I manage product information without ongoing developer help

Note: The main changes are made on the boolean filter as they are now radio buttons which should behave as intended compared to their checkbox counterpart.

## Related Issue

Closes #PWA-1689

## Verification Steps

Preconditions:
Web store with 2 store views created
Product attributes with input types Text, Content, Boolean, Multi-select and Date created in Admin>Store>Attributes>Product
Product attributes with Boolean and Multi-select inputs are enabled for layered navigation in Admin>Store>Attributes>Used in Layered Navigation
Product attributes  with Text, Content and Date inputs aren't available for using in layered navigation 
Products that use product attributes above are created in Admin Panel
User is browsing product listing page with products mentioned above on Venia PWA Storefront

Acceptance Criteria:
User navigates to the product listing page where the products belong to and sees product attributes enabled for the layered navigation positioned there based on rank set in Store>Product>Attribute>Storefront Properties>Position. If the rank is not set to define position based on it, product attributes sorted alphabetically.
User doesn't see attributes with input type Text, Content and Date as filters in the layered navigation 
User selects product attribute value from the options in the layered navigation and see product listing updated with the product that match the selected value
Multi-select attribute values are displayed in Layered navigation filters as designed 
Boolean attribute values are displayed in layered navigation as designed 

Test Plan:
Update backend with all needed pre-conditions and verify -
Text, Content and Date are not shown in layered nav (filter group)
Verify Multi select and boolean are displayed in layered nav.
Verify Applying multi select and boolean filters gets proper results.
Verify display order is based on by position by default. And if position is not set then it should be alphabetical order
Verify above on Store view 2
Verify localized labels works
Verify existing dropdown attribute still works
Verify above. on both Category. and Search pages.
Dont see any new UI so browser testing not needed. But if any new UI gets implemented like "Pre-order" label on product then do quick check on Firefox and Safari and a mobile device. Also check how date displays in Safari and firefox.

Cypress Test coverage:
We already have filter test . But for this ticket make sure graphql is automated fully in webapi functional tests.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run scaffold-pwa

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run pr-deploy

**Comment by michaelyu0123** [comment]
run lighthouse-desktop
run lighthouse-mobile

**Comment by michaelyu0123** [comment]
run lighthouse-desktop

**Comment by michaelyu0123** [comment]
run lighthouse-mobile

**Comment by michaelyu0123** [comment]
run lighthouse-mobile

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run all tests

**Review by jcalcaben — CHANGES_REQUESTED** [review]
Just a few missing entries for prop types, otherwise it looks good to me.

File: packages/venia-ui/lib/components/FilterModal/FilterList/filterItemRadio.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        toggleItem: func.isRequired,
+        removeGroup: func.isRequired
+    }).isRequired,
+    group: string.isRequired,
+    item: shape({
+        title: string.isRequired,
+        value: oneOfType([number, string]).isRequired,
+        label: string
+    }).isRequired,
+    onChange: func
```
[the remark this answers is not in the document]
Comment (michaelyu0123) [reply packages/venia-ui/lib/components/FilterModal/FilterList/filterItemRadio.js]: fixed both thanks

File: packages/venia-ui/lib/components/FilterModal/FilterList/filterItemRadioGroup.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }).isRequired,
+    filterState: setValidator,
+    group: string.isRequired,
+    items: arrayOf(
+        shape({
+            title: string.isRequired,
+            value: oneOfType([number, string]).isRequired
+        })
+    ).isRequired,
+    onChange: func
```
[the remark this answers is not in the document]
Comment (michaelyu0123) [reply packages/venia-ui/lib/components/FilterModal/FilterList/filterItemRadioGroup.js]: fixed both thanks

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run cypress

**Review by jcalcaben — DISMISSED** [review]

**Comment by michaelyu0123** [comment]
run all tests

**Comment by dpatil-magento** [comment]
@michaelyu0123  Please check following, other than these rest all looks good.
1. When there is one Boolean option for category then it still displaying as check box.
2. Cypress functional test added part this PR is failing.

**Comment by michaelyu0123** [comment]
yarn run all

**Comment by michaelyu0123** [comment]
yarn run tests

**Comment by michaelyu0123** [comment]
run all tests

**Comment by dpatil-magento** [comment]
QA Approved once the Cypress and lighthouse tests pass.

**Comment by dpatil-magento** [comment]
run cypress

**Review by jcalcaben — APPROVED** [review]

**Pull request merged**